src/client_side_request.cc

```diff
@@ -387,6 +387,9 @@ static void
 hostHeaderIpVerifyWrapper(const ipcache_addrs* ia, const Dns::LookupDetails &dns, void *data)
 {
     ClientRequestContext *c = static_cast<ClientRequestContext*>(data);
+    if (!c->httpStateIsValid())
```

Comment:
If these changes should be merged, please move them into a dedicated PR. In that PR, please discuss whether they fix any known/reproducible bugs and whether _all_ callbacks that should have this check now have it.

I am ignoring these changes in the other change requests.